core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -297,57 +313,107 @@ object ConsumerGroupCommand extends Logging {
       }.toArray
     }
 
-    private[admin] def collectGroupOffsets(): (Option[String], Option[Seq[PartitionAssignmentState]]) = {
-      val group = opts.options.valueOf(opts.groupOpt)
-      val consumerGroupSummary = adminClient.describeConsumerGroup(group, opts.options.valueOf(opts.timeoutMsOpt))
-      val assignments = consumerGroupSummary.consumers.map { consumers =>
-        var assignedTopicPartitions = Array[TopicPartition]()
-        val offsets = adminClient.listGroupOffsets(group)
-        val rowsWithConsumer =
-          if (offsets.isEmpty)
-            List[PartitionAssignmentState]()
-          else {
-            consumers.filter(_.assignment.nonEmpty).sortWith(_.assignment.size > _.assignment.size).flatMap { consumerSummary =>
-              val topicPartitions = consumerSummary.assignment
-              assignedTopicPartitions = assignedTopicPartitions ++ consumerSummary.assignment
-              val partitionOffsets: Map[TopicPartition, Option[Long]] = consumerSummary.assignment.map { topicPartition =>
-                new TopicPartition(topicPartition.topic, topicPartition.partition) -> offsets.get(topicPartition)
-              }.toMap
-              collectConsumerAssignment(group, Some(consumerGroupSummary.coordinator), topicPartitions,
-                partitionOffsets, Some(s"${consumerSummary.consumerId}"), Some(s"${consumerSummary.host}"),
-                Some(s"${consumerSummary.clientId}"))
-            }
-          }
+    def resetOffsets(): Map[TopicPartition, OffsetAndMetadata] = {
+      val groupId = opts.options.valueOf(opts.groupOpt)
+      val consumerGroups = adminClient.describeConsumerGroups(
+        util.Arrays.asList(groupId),
+        withTimeoutMs(new DescribeConsumerGroupsOptions)
+      ).describedGroups()
+
+      try {
+        val group = consumerGroups.get(groupId).get()
+        group.state.toString match {
+          case "Empty" | "Dead" =>
+            val partitionsToReset = getPartitionsToReset(groupId)
+            val preparedOffsets = prepareOffsetsToReset(groupId, partitionsToReset)
+
+            // Dry-run is the default behavior if --execute is not specified
+            val dryRun = opts.options.has(opts.dryRunOpt) || !opts.options.has(opts.executeOpt)
+            if (!dryRun)
+              getConsumer.commitSync(preparedOffsets.asJava)
+            preparedOffsets
+          case currentState =>
+            printError(s"Assignments can only be reset if the group '$groupId' is inactive, but the current state is $currentState.")
+            Map.empty
+        }
+      } catch {
+        case _: Throwable =>
+          printError(s"Coordinator not available for the group '$groupId'.")
```

Comment:
This is definitely one possible case, but are we sure exceptions can't be thrown in other cases? Also, we probably should pass the exception to `printError` rather than swallowing it.